src/components/common/CopyAddressButton/index.tsx

```diff
@@ -1,20 +1,27 @@
 import type { ReactNode, ReactElement } from 'react'
 import CopyButton from '../CopyButton'
+import CopyUntrustedAddressModal from './CopyUntrustedAddressModal'
 
 const CopyAddressButton = ({
   prefix,
   address,
   copyPrefix,
   children,
+  trusted,
 }: {
   prefix?: string
   address: string
   copyPrefix?: boolean
   children?: ReactNode
+  trusted?: boolean
 }): ReactElement => {
   const addressText = copyPrefix && prefix ? `${prefix}:${address}` : address
 
-  return <CopyButton text={addressText}>{children}</CopyButton>
+  return (
+    <CopyButton text={addressText} trusted={trusted} ConfirmationModal={CopyUntrustedAddressModal}>
```

Comment:
I would pass a Jsx element instead of a component function for the confirmation modal. And no need to have two props then. If a modal is passed = not trusted.

Reply:
A callback returning a JSX element would be another option, if you need some extra props from the parent component.